### PR TITLE
Modify ML part as fastforest is integrated within RBDT 

### PR DESCRIPTION
**Pull request by martamaja10**
Since fastforest was recently integrated with the RBDT library, the ML inference implementation is now modified. It is now not necessary to load fastforest library separately.

### CODE REVIEW

File: analyses/cms-open-data-ttbar/analysis.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def main() -> None:
         client = None
         load_cpp()
         if args.inference:
-            ml.load_cpp("./fastforest")
+            ml.load_cpp()
```
Comment (guitargeek) [inline analyses/cms-open-data-ttbar/analysis.py:351]:
```suggestion
```
You don't need this if statement at all now

**Review by guitargeek — APPROVED** [review]
Perfect!

File: analyses/cms-open-data-ttbar/analysis.py (hunk 1 above)
Comment (martamaja10) [reply analyses/cms-open-data-ttbar/analysis.py:351]: I still need this function to get some other ml helpers though

Comment (guitargeek) [reply analyses/cms-open-data-ttbar/analysis.py:351]: Oh, I see

File: analyses/cms-open-data-ttbar/analysis.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def parse_args() -> argparse.Namespace:
         help=(
             "Number of cores to use. In case of distributed execution this is the amount of cores per node."
         ),
-        default=len(os.sched_getaffinity(0)),
+        default=1,
```
Comment (vepadulano) [inline analyses/cms-open-data-ttbar/analysis.py:82]: Why this change in particular?

File: analyses/cms-open-data-ttbar/ml.py [hunk 3]
```diff
@@ -148,24 +138,24 @@ def define_features(df: ROOT.RDataFrame) -> ROOT.RDataFrame:
...
 
-
 def predict_proba(df: ROOT.RDataFrame) -> ROOT.RDataFrame:
     """get probability scores for every permutation in event"""
 
     # in inference, odd model applied to even events, while even model to odd events
     # read more about inference in dedicated part of AGC documentation:
     # https://agc.readthedocs.io/en/latest/taskbackground.html#machine-learning-component
+    
+    models_to_rootfiles()
```
Comment (vepadulano) [inline analyses/cms-open-data-ttbar/ml.py:148]: We could just decide to remove the need for this conversion step and just save the .root files directly. The conversion doesn't add anything relevant to the educational purposes of the analysis. Maybe it's not super important though

File: analyses/cms-open-data-ttbar/analysis.py [hunk 4]
```diff
@@ -358,7 +358,9 @@ def main() -> None:
             ROOT.RDF.Experimental.Distributed.initialize(load_cpp)
             if args.inference:
                 # TODO: make ml.load_cpp working on distributed
-                ROOT.RDF.Experimental.Distributed.initialize(ml.load_cpp, "./fastforest")
+                # ROOT.RDF.Experimental.Distributed.initialize(ml.load_cpp, "./fastforest")
```
Comment (vepadulano) [inline analyses/cms-open-data-ttbar/analysis.py:361]: Maybe remove the comment?

**Review by vepadulano — CHANGES_REQUESTED** [review]
Thanks for the nice improvements, some minor changes before merging. Also, I can see a few errors from the linter in the changed files such as `F401 os imported but unused` and `F541 f-string without any placeholders`

File: analyses/cms-open-data-ttbar/analysis.py (hunk 2 above)
Comment (martamaja10) [reply analyses/cms-open-data-ttbar/analysis.py:82]: the original one doesn't work on mac (this was just changed quickly, I can add something better thought over that works everywhere)

Comment (martamaja10) [reply analyses/cms-open-data-ttbar/analysis.py:82]: I can change it to: 
```
import multiprocessing
multiprocessing.cpu_count()
```

**Review by vepadulano — APPROVED** [review]
Thank you!
